src/System.Net.Security/src/System/Net/Security/SslState.cs

```diff
@@ -639,6 +639,13 @@ internal void ReplyOnReAuthentication(byte[] buffer)
         //
         private void ForceAuthentication(bool receiveFirst, byte[] buffer, AsyncProtocolRequest asyncRequest)
         {
+            if (asyncRequest != null && asyncRequest.CancellationToken.IsCancellationRequested)
+            {
+                // Cancel async operation, before I/O starts.
+                asyncRequest.CompleteUserWithError(new TaskCanceledException());
```

Comment:
Nit: instead of `new TaskCanceledException()`, it'd be better to do `new OperationCanceledException(asyncRequest.CancellationToken)`, so that the token that caused the cancellation is included in the exception.

Reply:
So I had that initially, and when I commented out all of the OperationCanceledException instance, and made cancellation throw from the InnerStream.WriteAsync/ReadAsync methods, it threw TaskCanceledException. Hence made these the same, to have uniform cancellation exception.